src/UI/Routing/Utils.ts

```diff
@@ -12,18 +12,26 @@ type NavigateTo = (
 
 /**
  * The useNavigateTo hook returns a navigateTo function which navigates to a route.
+ * @param newSearch This string should start with a question mark "?".
+ * @throws Will throw an error when newSearch is invalid
  */
 export const useNavigateTo = (): NavigateTo => {
   const { routeManager } = useContext(DependencyContext);
   const { search } = useLocation();
   const navigate = useNavigate();
 
   return (routeKind, params, newSearch) => {
+    if (newSearch !== undefined) validateSearch(newSearch);
     const pathname = routeManager.getUrl(routeKind, params);
-    navigate(`${pathname}?${newSearch || search}`);
+    navigate(`${pathname}${newSearch || search}`);
   };
 };
 
+const validateSearch = (search: string): void => {
+  if (search.startsWith("?")) return;
+  throw new Error("A search string should start with a question mark (?).");
```

Comment:
We could also add the question mark if it's not there

Reply:
I opted not to do that as the usage of `useNavigateTo` is completely in our hands.
We should use our own methods correctly. And when you forget it, an error is thrown.